File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int t) {
        int i = 0,j = nums.size() - 1;
        while(i<j){
            if((nums[i]+nums[j])>t){
                j--;
            }else if((nums[i]+nums[j])<t){
                i++;
            }else{
                return {i+1,j+1};
            }
        }
        return {i+1,j+1};
    }
};

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], t: int) -> int:
        low,high = 0,len(nums)-1

        while low <= high:
            mid = low + (high - low) // 2
            el = nums[mid]
            if el == t:
                return mid
            elif el < t:
                low = mid + 1
            else:
                high = mid - 1
            
        return -1
